contracts/protocol/facets/ExchangeHandlerFacet.sol

```diff
@@ -871,83 +951,105 @@ contract ExchangeHandlerFacet is IBosonExchangeHandler, BuyerBase, DisputeBase {
      * The buyer is allowed to commit if no group or condition is set for this offer.
      *
      * @param _buyer buyer address
-     * @param _offer the offer
-     * @param _exchangeId - the exchange id
+     * @param _condition - the condition to check
+     * @param _groupId - the group id
+     * @param _tokenId - the token id
      *
-     * @return bool - true if buyer is authorized to commit
+     * @return allow - true if buyer is authorized to commit
      */
-    function authorizeCommit(address _buyer, Offer storage _offer, uint256 _exchangeId) internal returns (bool) {
+    function authorizeCommit(
+        address _buyer,
+        Condition storage _condition,
+        uint256 _groupId,
+        uint256 _tokenId
+    ) internal returns (bool allow) {
         // Cache protocol lookups for reference
         ProtocolLib.ProtocolLookups storage lookups = protocolLookups();
 
-        // Allow by default
-        bool allow = true;
+        if (_condition.method == EvaluationMethod.SpecificToken) {
+            // If condition has a token id, check that the token id is in range, otherwise accept any token id
+            if (_condition.tokenId > 0) {
```

Comment:
I believe we should make this check based on `_condition.length > 0`.

It's a completely valid case that someone puts `_condition.tokenId = 0` and `_condition.length = 10` if they want to limit the range to the first 10 ids. But in this case, the range check would be skipped and someone with a token without

On the other hand, `_condition.length > 0` is a good indicator if the range needs to be validated or not.

Reply:
I used _condition. length > 1, because if length == 1 then we check _condition.tokenId and not _tokenId, which allow more use cases for conditional preminted vouchers